iroha/src/lib.rs

```diff
@@ -268,22 +269,29 @@ impl Iroha {
             }
         });
 
-        let queue = Arc::clone(&self.queue);
-        if let Some(ref genesis_block) = self.genesis_block {
-            log::info!("Initialize iroha using the genesis block");
-            for transaction in genesis_block.transactions.iter() {
-                if let Err(e) = queue
-                    .write()
-                    .await
-                    .push_pending_transaction(transaction.clone())
-                {
-                    log::error!(
-                        "Failed to push transactions from genesis block to queue error: {}",
-                        e
-                    )
+        let queue = &self.queue;
+        let sumeragi = &self.sumeragi;
+        self.wait_for_active_peers(&sumeragi.read().await.network_topology.sorted_peers)
```

Comment:
It is better not to call this directly but instead to spawn it as a task (`task::spawn`) and then put into `futures::join` section as it is done with other tasks in the initialization method.